### PR TITLE
Make noop async

### PR DESCRIPTION
**Pull request by fao89**
[noissue]

### CODE REVIEW

File: pulp_ansible/app/tasks/collections.py [hunk 1]
```diff
@@ -705,46 +704,42 @@ async def _find_all_collections(self):
...
-        root, api_version = await self._get_root_api(self.remote.url)
-        if api_version == 3:
-            downloader = self.remote.get_downloader(
-                url=root, silence_errors_for_response_status_codes={404}
-            )
-            try:
-                metadata = parse_metadata(await downloader.run())
-            except FileNotFoundError:
+        async with ProgressReport(message=msg, code="sync.no_change") as noop:
+            if not self.repository.remote:
```
Comment (dralley) [inline pulp_ansible/app/tasks/collections.py:708]: This and `.cast()` are likely to result in queries

File: pulp_ansible/app/tasks/collections.py [hunk 2]
```diff
@@ -705,46 +710,39 @@ async def _find_all_collections(self):
...
         """Check last synced metadata time."""
         msg = _("no_change: Checking if remote changed since last sync.")
-        noop = ProgressReport(message=msg, code="sync.no_change")
-        noop.state = TASK_STATES.COMPLETED
-        noop.save()
-
-        if not self.repository.remote:
-            return True
-
-        if self.remote != self.repository.remote.cast():
```
Comment (dralley) [inline pulp_ansible/app/tasks/collections.py:715]: https://github.com/pulp/pulpcore/pull/1614/files#diff-03bf88cd6f4c5051a7b1478ba05ea63287e7bb4c29feb00413b80ba35d22ccf2R75

File: pulp_ansible/app/tasks/collections.py [hunk 3]
```diff
@@ -705,46 +710,39 @@ async def _find_all_collections(self):
     async def _should_we_sync(self):
         """Check last synced metadata time."""
         msg = _("no_change: Checking if remote changed since last sync.")
-        noop = ProgressReport(message=msg, code="sync.no_change")
-        noop.state = TASK_STATES.COMPLETED
-        noop.save()
-
-        if not self.repository.remote:
-            return True
```
Comment (dralley) [inline pulp_ansible/app/tasks/collections.py:713]: Same here, I don't know if the refactor was for other reasons but if it was just to resolve this issue you could do something like `await sync_to_async(self.repository.remote)` (might need to be wrapped in a function?)

Comment (fao89) [reply pulp_ansible/app/tasks/collections.py:713]: I decided to move it out of the async code

**Review by bmbouter — APPROVED** [review]
This would probably benefit from a refactor issue attached to this PR in case it needs a backport.

I probably would have kept it structured as-is and converted the calls to be async safe, but I'm ok with doing it this way too.

**Pull request merged**